### ISSUE TITLE
Query batching breaks findUnique using Bytes

### ISSUE THREAD

**Issue by MarcGuiselin** [prisma/prisma#10604]
### Bug description

It seems under certain circumstances the Prisma client library is incorrectly interpreting the Byte type as a base64 string. For example in prisma studio, trying to delete any record in the `Table` (see the schema below) table will fail. See the related Prisma Studio issue: https://github.com/prisma/studio/issues/804

If you try to fetch two table rows at the same time, while searching on a `Bytes` field, prisma will throw the following error: `Have: Object({"in": List([String(base64), String(base64)])}), want: Bytes]`

[The docs say that findUnique queries are batched together](https://www.prisma.io/docs/guides/performance-and-optimization/query-optimization-performance#solving-n1-in-graphql-with-findunique-and-prismas-dataloader): In this case, I suspect that the Prisma client library is batching requests that it shouldn't be. The error shows that the two queries are being batched together into something alike the following invalid query:

```js
where: {
  field: {
    in: [Buffer.from('Hello World'), Buffer.from('Hello World')]
  },
},
```

### How to reproduce

Run the following node.js code:

```js
async function main(){
  await Promise.all([
    prisma.table.findUnique({
      where: {
        field: Buffer.from('Hello World') // some buffer 1
      },
    }),
    prisma.table.findUnique({
      where: {
        field: Buffer.from('Hello World') // same or different from buffer 1
      },
    }),
  ])
}
main()
```

You'll see the following error:

```
Failed to validate the query: `Unable to match input value to any allowed input type for the field. Parse errors: [Query parsing/validation error at `Query.findManyTable.where.TableWhereInput.field.BytesFilter.in`: Field does not exist on enclosing type., Query parsing/validation error at `Query.findManyTable.where.TableWhereInput.field`: Value types mismatch. Have: Object({"in": List([String("SGVsbG8gV29ybGQ="), String("SGVsbG8gV29ybGQ=")])}), want: Bytes]` at `Query.findManyTable.where.TableWhereInput.field`
```

### Expected behavior

There shouldn't be an error. These queries were batched together when they shouldn't be.

### Prisma information

```prisma
datasource db {
  provider = "postgres"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

model Table {
  field Bytes @id
}
```

### Environment & setup

- OS: Ubuntu
- Database: PostgreSQL,
- Node.js version: v17.1.0

### Prisma Version

```
prisma                  : 3.0.2
@prisma/client          : 3.0.2
Current platform        : debian-openssl-1.1.x
Query Engine (Node-API) : libquery-engine 2452cc6313d52b8b9a96999ac0e974d0aedf88db (at ../../node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
Migration Engine        : migration-engine-cli 2452cc6313d52b8b9a96999ac0e974d0aedf88db (at ../../node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
Introspection Engine    : introspection-core 2452cc6313d52b8b9a96999ac0e974d0aedf88db (at ../../node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x)
Format Binary           : prisma-fmt 2452cc6313d52b8b9a96999ac0e974d0aedf88db (at ../../node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x)
Default Engines Hash    : 2452cc6313d52b8b9a96999ac0e974d0aedf88db
Studio                  : 0.423.0
```

**Comment by janpio** [comment]
Can you possible try with a more recent Prisma version to eliminate the possibility that this was already fixed in a newer version of Prisma?

**Comment by MarcGuiselin** [comment]
> Can you possible try with a more recent Prisma version to eliminate the possibility that this was already fixed in a newer version of Prisma?

Wow, I was more than a release or two behind. My bad.

This has been fixed as of Prisma Client v3.6.0. However https://github.com/prisma/studio/issues/804 persists.
```
prisma                  : 3.6.0
@prisma/client          : 3.6.0
Current platform        : debian-openssl-1.1.x
Query Engine (Node-API) : libquery-engine dc520b92b1ebb2d28dc3161f9f82e875bd35d727 (at node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
Migration Engine        : migration-engine-cli dc520b92b1ebb2d28dc3161f9f82e875bd35d727 (at node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
Introspection Engine    : introspection-core dc520b92b1ebb2d28dc3161f9f82e875bd35d727 (at node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x)
Format Binary           : prisma-fmt dc520b92b1ebb2d28dc3161f9f82e875bd35d727 (at node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x)
Default Engines Hash    : dc520b92b1ebb2d28dc3161f9f82e875bd35d727
Studio                  : 0.440.0
```
